nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -304,6 +304,113 @@ in {
           phone-numbers.
         '';
       };
+
+      objectstore = let
+        s3Arguments = {
+          bucket = mkOption {
+            type = types.str;
+            example = "nextcloud";
+            description = ''
+              The name of the S3 bucket.
+            '';
+          };
+          autocreate = mkOption {
+            type = types.bool;
+            description = ''
+              Create the objectstore if it does not exist.
+            '';
+          };
+          key = mkOption {
+            type = types.str;
+            example = "EJ39ITYZEUH5BGWDRUFY";
+            description = ''
+              The access key for the S3 bucket.
+            '';
+          };
+          secretFile = mkOption {
+            type = types.str;
+            example = "/var/nextcloud-objectstore-s3-secret";
+            description = ''
+              The full path to a file that contains the access secret. Must be
+              readable by user <literal>nextcloud</literal>.
+            '';
+          };
+          hostname = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            example = "example.com";
+            description = ''
+              Required for some non-Amazon implementations.
+            '';
+          };
+          port = mkOption {
+            type = types.nullOr types.port;
+            default = null;
+            description = ''
+              Required for some non-Amazon implementations.
+            '';
+          };
+          useSsl = mkOption {
+            type = types.nullOr types.bool;
```

Comment:
We either use SSL or we don't, correct? :) 
(and are there any reasons for turning it off? It should be `true` at least).

Reply:
The difference between `null` and us deciding on a default value is that in the case the value is `null`, we don't add a `use_ssl` line to the config at all and in turn fall back to whatever Nextcloud's default happens to be.

That said I understand if you'd prefer we try to encourage the use of SSL - I'm happy to change the default to `true`.

Reply:
> we don't add a use_ssl line to the config at all and in turn fall back to whatever Nextcloud's default happens to be

Can't we just use the fallback that Nextcloud uses (i.e. Nextcloud's default is our default). Having `null` always makes a special-case mandatory which isn't really nice.

Reply:
Sure, it's not documented anywhere but [judging by the src](https://github.com/nextcloud/server/blob/e2116e2fb226890341c548e3f7d79c0ac63c1b06/lib/private/Files/ObjectStore/S3ConnectionTrait.php#L83) it looks like SSL is enabled by default.